sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient.cs

```diff
@@ -0,0 +1,969 @@
+            try
+            {
+                var response = RestClient.GroupExists(hub, group, cancellationToken);
+                return Response.FromValue(response.Status == 200, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>Check if there are any client connections connected for the given user</summary>
+        /// <param name="userId">Target user Id</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public virtual async Task<Response<bool>> UserExistsAsync(string userId, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("WebPubSubServiceRestClient.CheckUserExistenceAsync");
+            scope.Start();
+            try
+            {
+                var response = await RestClient.UserExistsAsync(hub, userId, cancellationToken).ConfigureAwait(false);
+                return Response.FromValue(response.Status == 200, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>Check if there are any client connections connected for the given user</summary>
+        /// <param name="userId">Target user Id</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Response"/> representing the result of the operation.</returns>
+        public virtual Response<bool> UserExists(string userId, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("WebPubSubServiceRestClient.CheckUserExistence");
+            scope.Start();
+            try
+            {
+                var response = RestClient.UserExists(hub, userId, cancellationToken);
+                return Response.FromValue(response.Status == 200, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>Add a connection to the target group.</summary>
+        /// <param name="group">Target group name, which length should be greater than 0 and less than 1025.</param>
+        /// <param name="connectionId">Target connection Id</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public virtual async Task<Response> AddConnectionToGroupAsync(string group, string connectionId, CancellationToken cancellationToken = default)
+        {
+            if (connectionId is null)
+            {
+                throw new ArgumentNullException(nameof(connectionId));
+            }
```

Comment:
- Why do we validate this but not the group?  Do we need to validate it?
- Prefer `Argument.IsNotNull(connectionId, nameof(connectionId))` for consistency